data/src/main/java/com/teamagam/gimelgimel/data/geometry/repository/inMemory/InMemoryGeoDataCache.java

```diff
@@ -0,0 +1,72 @@
+package com.teamagam.gimelgimel.data.geometry.repository.inMemory;
+
+import com.teamagam.gimelgimel.domain.geometries.entities.GeoEntity;
+import com.teamagam.gimelgimel.domain.geometries.entities.VectorLayer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import rx.Observable;
+
+public class InMemoryGeoDataCache {
+    private List<VectorLayer> mVectorLayers;
+
+    public InMemoryGeoDataCache() {
+        mVectorLayers = new ArrayList<>();
+    }
+
+    public Observable<VectorLayer> getAllVectorLayers() {
+        return Observable.from(mVectorLayers);
+    }
+
+    public Observable<VectorLayer> getVectorLayerById(String id) {
+        return Observable.from(mVectorLayers)
+                .filter(vectorLayer -> vectorLayer.getLayerId().equals(id));
+    }
+
+    public Observable<GeoEntity> getEntity(String layerId, String entityId) {
+        return getVectorLayerById(layerId)
+                .map(vectorLayer -> vectorLayer.getEntity(entityId));
+    }
+
+    public void addVectorLayer(VectorLayer vectorLayer) {
+        mVectorLayers.add(vectorLayer);
+    }
+
+    public void addVectorLayers(VectorLayer[] vectorLayers) {
+        Observable.from(vectorLayers)
+                .subscribe(vectorLayer -> mVectorLayers.add(vectorLayer));
+    }
+
+
+    public void addGeoEntityToVectorLayer(String layerId, GeoEntity geoEntity) {
+        getVectorLayerById(layerId)
+                .subscribe(vectorLayer -> vectorLayer.addEntity(geoEntity));
+    }
+
+    public void addGeoEntitiesToVectorLayer(String layerId, GeoEntity[] geoEntities) {
+        getVectorLayerById(layerId)
+                .subscribe(vectorLayer ->
+                        Observable.from(geoEntities)
```

Comment:
why are adding the entities one by one? the vector layer data should support adding collection.